File: SlopeFreqResp.py
import numpy as np
import control.matlab as cm


def SlopeFreqResp(Sys=None, Omega=None):
    # j = Imaginary unit sqrt(-1) identify in python as 1j
    FR = cm.evalfr(Sys, np.dot(1j, Omega))
    # SlopeFreqResp.m:2
    Omega1 = np.dot(0.99, Omega)
    # SlopeFreqResp.m:3
    FR1 = cm.evalfr(Sys, np.dot(1j, Omega1))
    # SlopeFreqResp.m:3
    Omega2 = np.dot(1.01, Omega)
    # SlopeFreqResp.m:4
    FR2 = cm.evalfr(Sys, np.dot(1j, Omega2))
    # SlopeFreqResp.m:4
    dFR = (FR2 - FR1) / (Omega2 - Omega1)
    # SlopeFreqResp.m:5
    return FR, dFR

File: SlopeFreqResp.m
function [FR,dFR]=SlopeFreqResp(Sys,Omega);
FR=evalfr(Sys,j*Omega);
Omega1=0.99*Omega; FR1=evalfr(Sys,j*Omega1);
Omega2=1.01*Omega; FR2=evalfr(Sys,j*Omega2);
dFR=(FR2-FR1)/(Omega2-Omega1);
end